extensions/VisualEditor/wikia/modules/ve/dm/ve.dm.WikiaTransclusionModel.js

```diff
@@ -91,16 +116,11 @@ ve.dm.WikiaTransclusionModel.prototype.fetchInfoboxParamsRequestDone = function
 				};
 			}
 
-			for ( i = 0; i < page.infoboxes.length; i++ ) {
-				page.infoboxes[i].sources = Object.keys( page.infoboxes[i].sourcelabels );
-
-				for ( j = 0; j < page.infoboxes[i].sources.length; j++ ) {
-					source = page.infoboxes[i].sources[j];
-
-					specs[page.title].params[ source ] = { label: page.infoboxes[i].sourcelabels[source] };
-					specs[page.title].paramOrder.push( page.infoboxes[i].sources[j] );
-				}
-			}
+			page.infoboxes.forEach( function ( infobox ) {
+				infobox.metadata.forEach( function ( node ) {
+					parseInfoboxMetadata( node, specs[page.title] );
```

Comment:
`page.infoboxes` has as many elements as there are infoboxes on the page, usually 0 or 1.
`infobox.metadata` has as many elements as there are first level nodes in the infobox.
`parseInfoboxMetadata` is recursive and calls itself when the node is a group.

It's not really O(n3), more like O(n * m * unknown depth) :wink: 
In the real life cases these shouldn't get big enough to be worried about complexity here.